examples/flutter_gallery/lib/demo/expansion_panels_demo.dart

```diff
@@ -0,0 +1,328 @@
+  Widget build(BuildContext context) {
+    return new DefaultTextStyle(
+      style: new TextStyle(
+        color: Colors.black54,
+        fontSize: 15.0
+      ),
+      child: new Column(
+        children: <Widget>[
+          new Container(
+            margin: const EdgeInsets.only(
+              left: 24.0,
+              right: 24.0,
+              bottom: 24.0
+            ) - margin,
+            child: new Center(
+              child: child
+            )
+          ),
+          new Container(
+            padding: const EdgeInsets.symmetric(vertical: 16.0),
+            decoration: new BoxDecoration(
+              border: new Border(
+                top: new BorderSide(color: Theme.of(context).dividerColor)
+              )
+            ),
+            child: new Row(
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: <Widget>[
+                new Container(
+                  margin: const EdgeInsets.only(right: 8.0),
+                  child: new FlatButton(
+                    onPressed: onCancel,
+                    child: new DefaultTextStyle(
+                      style: new TextStyle(
+                        color: Colors.black54,
+                        fontSize: 15.0,
+                        fontWeight: FontWeight.w500
+                      ),
+                      child: new Text('CANCEL')
+                    )
+                  )
+                ),
+                new Container(
+                  margin: const EdgeInsets.only(right: 8.0),
+                  child: new FlatButton(
+                    onPressed: onSave,
+                    textTheme: ButtonTextTheme.accent,
+                    child: new Text('SAVE')
+                  )
+                )
+              ]
+            )
+          )
+        ]
+      )
+    );
+  }
+}
+
+class ExpasionPanelsDemo extends StatefulWidget {
+  static const String routeName = '/expansion_panels';
+
+  @override
+  _ExpansionPanelsDemoState createState() => new _ExpansionPanelsDemoState();
+}
+
+class _ExpansionPanelsDemoState extends State<ExpasionPanelsDemo> {
+  String _tripName = 'Caribbean cruise';
+  String get tripName => _tripName;
+  _Location _location = _Location.Barbados;
+  String get location => _location.toString().split(".")[1];
+  double _sunAmount = 80.0;
+  String get sunAmount => '${_sunAmount.round()}';
+  List<bool> expanded = new List<bool>.filled(3, false);
+
+  Widget _body(Widget child, int index, {EdgeInsets margin: EdgeInsets.zero}) {
+    return new CollapsibleBody(
+      margin: margin,
+      child: child,
+      onSave: () {
+        setState(() {
+          expanded[index] = false;
+        });
+      },
+      onCancel: () {
+        setState(() {
+          expanded[index] = false;
+        });
+      }
+    );
+  }
+
+  void _changeLocation(_Location newLocation) {
+    setState(() {
+      _location = newLocation;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
The way this is constructed could be less fragile. Imagine inserting another expansion panel or deleting. The expanded array and its indices need to be updated in many places. 

It seems like an aspect of the list of panels, probably in terms of their expanded state and the contents of their headers, should be factored out.